File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
import java.util.function.IntUnaryOperator;

class ArmstrongNumbers {
	boolean isArmstrongNumber(int numberToCheck) {
		String numStr = String.valueOf(numberToCheck);
		int numberOfDigits = numStr.length();

		int sum = numStr.chars()
				          .map(Character::getNumericValue)
				          .map(this.raiseToPower(numberOfDigits))
				          .sum();

		return sum == numberToCheck;
	}

	private IntUnaryOperator raiseToPower(int numberOfDigits) {
		return digit -> (int) Math.pow(digit, numberOfDigits);
	}
}

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def get_digits(number):
    return [int(digit) for digit in str(number)]

def get_armstrong_sum(number):
    digits = get_digits(number)
    number_of_digits = len(digits)
    return sum(digit ** number_of_digits for digit in digits)

def is_armstrong_number(number):
    return get_armstrong_sum(number) == number